### PR TITLE
[DPE-5324] Increase linting rules

### PR DESCRIPTION
**Pull request by dragomirp**
Add more linting rules and change and suppress the reported issues. Added rules are:
* [flake8-bugbear](https://docs.astral.sh/ruff/rules/#flake8-bugbear-b)
* [ruff specific rules](https://docs.astral.sh/ruff/rules/#ruff-specific-rules-ruf)
* [flake8-bandit](https://docs.astral.sh/ruff/rules/#flake8-bandit-s)
* [flake8-simplify](https://docs.astral.sh/ruff/rules/#flake8-simplify-sim)
* [pyupdate](https://docs.astral.sh/ruff/rules/#pyupgrade-up)
* [flake8-type-checking](https://docs.astral.sh/ruff/rules/#flake8-type-checking-tch)

### CODE REVIEW

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 1]
```diff
@@ -173,48 +172,55 @@ def create_database(
...
         """
+        plugins = plugins if plugins else []
+        client_relations = client_relations if client_relations else []
         try:
             connection = self._connect_to_database()
             cursor = connection.cursor()
-            cursor.execute(f"SELECT datname FROM pg_database WHERE datname='{database}';")
+            cursor.execute(
+                SQL("SELECT datname FROM pg_database WHERE datname={};").format(Literal(database))
+            )
```
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:182]: Bandit checks don't like rendering directly in the SQL string.

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 2]
```diff
@@ -109,14 +108,14 @@ def __init__(
         user: str,
         password: str,
         database: str,
-        system_users: List[str] = [],
+        system_users: Optional[list[str]] = None,
```
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:111]: We can't use implicit union (`list[str] | None`) for compatibility with focal (Python 3.8)

File: src/rotate_logs.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 def main():
     """Main loop that calls logrotate."""
     while True:
-        subprocess.run(["logrotate", "-f", "/etc/logrotate.d/pgbackrest.logrotate"])
+        # Command is hardcoded
+        subprocess.run(["/usr/sbin/logrotate", "-f", "/etc/logrotate.d/pgbackrest.logrotate"])  # noqa: S603
 
         # Wait 60 seconds before executing logrotate again.
         time.sleep(60)
```
Comment (dragomirp) [inline src/rotate_logs.py:17]: We can most probably use pebble backoff to do the same

File: tests/unit/test_async_replication.py [hunk 4]
```diff
@@ -216,6 +221,10 @@ def test_on_async_relation_changed(harness, wait_for_standby):
             "relations.async_replication.PostgreSQLAsyncReplication._wait_for_standby_leader",
             return_value=wait_for_standby,
         ),
+        patch(
+            "relations.async_replication.PostgreSQLAsyncReplication._get_unit_ip",
+            return_value="1.1.1.1",
+        ) as _get_unit_ip,
```
Comment (dragomirp) [inline tests/unit/test_async_replication.py:227]: Fails on non-Ubutu hosts without the patch.

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 5]
```diff
@@ -382,42 +390,38 @@ def _generate_database_privileges_statements(
...
-                    sql.Identifier(user),
-                    sql.Identifier(user),
-                    sql.Identifier(user),
+                    Identifier(user),
+                    Identifier(user),
+                    Identifier(user),
+                    Identifier(user),
                 )
             )
-            statements.append(
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply lib/charms/postgresql_k8s/v0/postgresql.py:391]: Likely removed during merging. Restored.

**Review by marceloneppel — APPROVED** [review]
Thanks for the great work, Dragomir!

**Pull request merged**